fiat-google-groups/src/test/groovy/com/netflix/spinnaker/fiat/roles/google/GoogleDirectoryUserRolesProviderSpec.groovy

```diff
@@ -80,7 +80,77 @@ class GoogleDirectoryUserRolesProviderSpec extends Specification {
 
     }
 
+    def "should recursively collect all nested groups"() {
+        given:
+        def provider = Spy(GoogleDirectoryUserRolesProvider) {
+            // Mock the recursive group relationships
+            expandIndirectGroups() >> true
+
+            getGroupsFromEmail("root@example.com") >> new Groups(groups: [
+                new Group(email: "child1@example.com"),
+                new Group(email: "child2@example.com")
+            ])
+
+            getGroupsFromEmail("child1@example.com") >> new Groups(groups: [
+                new Group(email: "grandchild1@example.com")
+            ])
+
+            getGroupsFromEmail("child2@example.com") >> new Groups(groups: [
+                new Group(email: "grandchild2@example.com"),
+                new Group(email: "child1@example.com") // Cycle to test deduplication
+            ])
+
+            getGroupsFromEmail("grandchild1@example.com") >> new Groups(groups: [])
+            getGroupsFromEmail("grandchild2@example.com") >> null
+            }
+        when:
+        def result = provider.getGroupsFromEmailRecursively("root@example.com")
+
+        then:
+        result.groups*.email.containsAll([
+            "child1@example.com",
+            "child2@example.com",
+            "grandchild1@example.com",
+            "grandchild2@example.com"
+            ])
+        result.groups.size() == 4
+    }
+
+        def "should not recursively collect all nested groups"() {
```

Comment:
There's extra whitespace, and a lot of duplication here.  Can you try parameterizing the previous test?  Something like annotating with `@Unroll`, adding
```
where:
expandIndirectGroups << [false, true]
```
and adding `#expandIndirectGroups` to the test name and logic.